test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -295,8 +295,15 @@ protected static Optional<Boolean> clusterHasCapability(
         if (capabilities.isEmpty() == false) {
             request.addParameter("capabilities", String.join(",", capabilities));
         }
-        Map<String, Object> response = entityAsMap(client.performRequest(request).getEntity());
-        return Optional.ofNullable((Boolean) response.get("supported"));
+        try {
+            Map<String, Object> response = entityAsMap(client.performRequest(request).getEntity());
+            return Optional.ofNullable((Boolean) response.get("supported"));
```

Comment:
Hmm if we get `null` here that means that some node failed to respond. That seems like it should be a test failure? I.e. can we assert that this isn't `null`?

Reply:
Wondering, @thecoop, wouldn't it be more consistent if the API returned a server error in this case?

Reply:
I'll keep this as is for now. We're going to discuss this independently as a follow up item.